### DISCUSSION TITLE
Consider implementing template fragments in Blade

### DISCUSSION THREAD

**Discussion by panda-madness** [laravel/framework#44468] [Ideas] [answered]
With the increasing popularity of various server render driven frontend frameworks (Turbo, Htmx, Unpoly to name a few) one feature that would make working with these types of frameworks much easier would be so called template fragments. [A detailed description is available here](https://htmx.org/essays/template-fragments/).

In short, it's a feature that allows us the ability to render only a fragment of a given template instead of the whole thing, without having to split out templates into many small files.

Given a blade template:
```blade
<div>
    <div>Post title</div>
    @fragment('status_controls')
        <div>Archived</div>
        <div>
            <button>Unarchive</button>
            <button>Delete</button>
        </div>
    @endfragment
    
    <div>Some content</div>
</div>
```
We could render the whole template on page load, and only the contents of the defined fragment on subsequent requests:
```php
view('post')->fragment('status_controls');
```

**Comment by brunoalod** [comment]
Submitted a PR for this. #44774

**Comment by brunoalod** [comment, marked as answer]
It got merged 😃

**Reply by panda-madness** [reply]
Love you, mate!
